src/libraries/System.Threading.Tasks.Dataflow/tests/Dataflow/TransformManyBlockTests.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,663 @@
+                                    MaxDegreeOfParallelism = dop,
+                                    MaxMessagesPerTask = maxMessagesPerTask,
+                                    TaskScheduler = scheduler
+                                };
+                                TransformManyBlock<int, int> tb = new TransformManyBlock<int, int>(i => AsyncEnumerable.Repeat(i, elementsPerItem), options);
+
+                                await Task.WhenAll(
+                                    Task.Run(async delegate
+                                    { // consumer
+                                        int i = 0;
+                                        int processed = 0;
+                                        while (await tb.OutputAvailableAsync())
+                                        {
+                                            Assert.Equal(expected: i, actual: await tb.ReceiveAsync());
+                                            processed++;
+                                            if (processed % elementsPerItem == 0)
+                                            {
+                                                i++;
+                                            }
+                                        }
+                                    }),
+                                    Task.Run(async delegate
+                                    { // producer
+                                        for (int i = 0; i < Messages; i++)
+                                        {
+                                            await tb.SendAsync(i);
+                                        }
+                                        tb.Complete();
+                                    }));
+                            }
+        }
+
+        [Fact]
+        public async Task TestMessagePostponementAsyncEnumerable()
+        {
+            const int Excess = 10;
+            foreach (int boundedCapacity in new[] { 1, 3 })
+            {
+                var options = new ExecutionDataflowBlockOptions { BoundedCapacity = boundedCapacity };
+                var tb = new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable, options);
+
+                var sendAsync = new Task<bool>[boundedCapacity + Excess];
+                for (int i = 0; i < boundedCapacity + Excess; i++)
+                {
+                    sendAsync[i] = tb.SendAsync(i);
+                }
+                tb.Complete();
+
+                for (int i = 0; i < boundedCapacity; i++)
+                {
+                    Assert.True(sendAsync[i].IsCompleted);
+                    Assert.True(sendAsync[i].Result);
+                }
+
+                for (int i = 0; i < Excess; i++)
+                {
+                    Assert.False(await sendAsync[boundedCapacity + i]);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task TestMultipleYieldsAsyncEnumerable()
+        {
+            const int Messages = 10;
```

Comment:
```suggestion
            const int messages = 10;
```
private constants should be written in lower case